vendor/github.com/bitrise-io/go-xcode/v2/xcodecache/swiftpm_cache.go

```diff
@@ -35,11 +36,13 @@ func (c swiftPackageCache) SwiftPackagesPath(xcodeProjectPath string) (string, e
 		return "", fmt.Errorf("project path not an absolute path: %s", xcodeProjectPath)
 	}
 
-	if !strings.HasSuffix(xcodeProjectPath, ".xcodeproj") && !strings.HasSuffix(xcodeProjectPath, ".xcworkspace") {
-		return "", fmt.Errorf("invalid Xcode project path %s, no .xcodeproj or .xcworkspace suffix found", xcodeProjectPath)
+	fileExtension := filepath.Ext(xcodeProjectPath)
+	if fileExtension != ".xcodeproj" && fileExtension != ".xcworkspace" && fileExtension != ".swift" {
+		return "", fmt.Errorf("invalid Xcode project path: %s, no .xcodeproj or .xcworkspace suffix, or Package.swift file found", xcodeProjectPath)
 	}
 
-	projectDerivedData, err := xcodeProjectDerivedDataPath(xcodeProjectPath)
+	trimmedXcodeProjectPath := strings.TrimSuffix(xcodeProjectPath, "/Package.swift")
```

Comment:
Yes, the current implementation only caches the `SourcePackages` in DerivedData, I'm not sure if it's still relevant or useful. Did you fine a trustworthy source about what each cache folder is about and which one is a good practice to cache in CI?

Reply:
I'll create a follow-up ticket about this, I also found a bug in the current cache implementation. We need to revisit and re-test our approach, but this will be out of scope for this PR.